### PR TITLE
Switch to all-in-1/selfcontained docker builds and GH actions

### PR DESCRIPTION
**Pull request by hoerup**
Use github actions to auto build and publish images to ghcr.io container image repo

### CODE REVIEW

**Comment by sana-jawad** [comment]
@hoerup thanks for the contribution. Have you tested this change?

**Comment by hoerup** [comment]
the dockerimage it self is ok 
actions are a bit tricky to get working on a fork

**Comment by sana-jawad** [comment]
Hi @hoerup I couldn't get to review it this week. Our team will review it in coming week. Apologies for the delay.

File: Dockerfile [hunk 1]
```diff
@@ -1,11 +1,20 @@
...
 RUN curl -o /aws-iam-authenticator https://amazon-eks.s3-us-west-2.amazonaws.com/1.14.6/2019-08-22/bin/linux/amd64/aws-iam-authenticator \
   && wait \
   && chmod +x /aws-iam-authenticator
 
+COPY . /build/
+WORKDIR /build
+
+RUN go env -w GO111MODULE=auto \
+   && make
+
```
Comment (sana-jawad) [inline Dockerfile:12]: extra lines?

File: .github/workflows/docker.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+
+      - name: Build container image
+        uses: docker/build-push-action@v2
+        with:
+          push: true
+          platforms: linux/amd64,linux/arm64
```
Comment (sana-jawad) [inline .github/workflows/docker.yml:41]: can we please add darwin/amd64, windows/amd64' in platforms?

File: .github/workflows/docker.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+          file: Dockerfile
+          tags:  ${{ steps.meta.outputs.tags  }}
+
+
```
Comment (sana-jawad) [inline .github/workflows/docker.yml:45]: nit:please remove extra lines.

**Comment by sana-jawad** [comment]
> merge this - you nee



> the dockerimage it self is ok actions are a bit tricky to get working on a fork, but i created a new repo and pushed my changes there
> 
> you can see the result here https://github.com/hoerup/sloop-copy/actions/runs/2035949576
> 
> and published to https://github.com/hoerup/sloop-copy/pkgs/container/sloop-copy
> 
> Update: if/when you merge this - you need to update the new package repo and set it to public available

The changes look good to me. Would it be possible to run it again as the log lines are not visible now?
Also can you please elaborate "if/when you merge this - you need to update the new package repo and set it to public available"?

**Review by sana-jawad — APPROVED** [review]

**Pull request merged**